### PR TITLE
Re-write readonly editions

### PR DESCRIPTION
**Pull request by stijndcl**
- Made `read-only` a field of the edition instead of saying only the latest edition is editable
- Made a new route to change the status of an edition (admin-only)
- Sorted editions by year descending, and then by id
- Fix a few bugs in frontend
- Made an easy & reliable function to check if an edition is read-only or not (so we can hide components in frontend dynamically)
- Made an interactive label that displays the state of an edition on the `/editions` page. Can be clicked to toggle the state. This can _not_ be clicked by coaches, and is only visual to them. This label has no confirmation because it's a non-destructive action that can be undone in just 1 click.

## Note
I've had to re-write quite a bit of stuff to get this working, as lots of systems work together. It would be really cool if someone could quickly stress-test this a bit & double-check if everything still works. I've patched about 20 small edge cases myself, and I think I've got all of them now.

### CODE REVIEW

**Review by BertGuillemyn — COMMENTED** [review]
I've had a look at the backend part, and everything looks fine for me there. The frontend also seems ok for me, but maybe someone else needs to take a look at this. 
I've tried some things out on the actual site, and everything still seems to be working as expected. Maybe it's not super intuitive that clicking on the 'editable' label will make it read-only, but as you stated, you can revert this immediately by clicking on the label again. 
I think this will give some conflicts with #433, maybe we should merge that one first and then add the extra checks here as well?

**Comment by stijndcl** [comment]
> I've tried some things out on the actual site, and everything still seems to be working as expected. Maybe it's not super intuitive that clicking on the 'editable' label will make it read-only, but as you stated, you can revert this immediately by clicking on the label again. 

Hmm I can get behind that, but I found it difficult to think of a nice way to do this, considering we only have like 2 days left.

> I think this will give some conflicts with #433, maybe we should merge that one first and then add the extra checks here as well?

Yes good point

**Review by BertGuillemyn — APPROVED** [review]

**Review by SeppeM8 — APPROVED** [review]
Couldn't find any bugs in frontend. (Projects are not tested yet since they are not yet on Develop.
Well done.

File: backend/src/app/routers/editions/editions.py [hunk 1]
```diff
@@ -45,6 +45,17 @@ async def get_editions(db: AsyncSession = Depends(get_session), user: User = Dep
     return EditionList(editions=user.editions)
 
 
+@editions_router.patch("/{edition_name}", response_class=Response, tags=[Tags.EDITIONS],
+                       dependencies=[Depends(require_admin)], status_code=status.HTTP_204_NO_CONTENT)
+async def patch_edition(edit_edition: EditEdition, edition: EditionDB = Depends(get_edition),
```
Comment (FKD13) [inline backend/src/app/routers/editions/editions.py:50]: `Depends(get_edition)` Does this mean you can edit a readonly edition ?!?!?!!? :upside_down_face:

File: backend/src/app/routers/login/login.py [hunk 2]
```diff
@@ -61,7 +62,8 @@ async def generate_token_response_for_user(db: AsyncSession, user: User) -> Toke
     access_token, refresh_token = create_tokens(user)
 
     user_data: dict = user_model_to_schema(user).__dict__
-    user_data["editions"] = await get_user_editions(db, user)
+    editions = await get_user_editions(db, user)
+    user_data["editions"] = list(map(Edition.from_orm, editions))
```
[the remark this answers is not in the document]
Comment (stijndcl) [reply backend/src/app/routers/login/login.py:66]: Unfortunately it can't, because of pydantic-related limitations :/

**Review by FKD13 — APPROVED** [review]
